### PR TITLE
Update max_model_len calculation and fixup encoder pooling

### PR DESCRIPTION
**Pull request by Datta0**
**What this PR does / why we need it**: 

[vLLM updated their max_model_len](https://github.com/vllm-project/vllm/blob/2ac6d0e75bc846998da56b50bf4f8853cb36d484/vllm/config.py#L1870) calculations and hence we'd want to use the same. 
Also, `_mean_pooling` wasn't always happening on the right device. so fix it.

### CODE REVIEW

File: python/huggingfaceserver/huggingfaceserver/utils.py [hunk 1]
```diff
@@ -13,19 +13,25 @@
...
 
 from kserve.logging import logger
 
 
+ALLOW_LONG_MAX_MODEL_LEN = "ALLOW_LONG_MAX_MODEL_LEN"
+
+
 # "This implementation is based on vLLM's _get_and_verify_max_len "
-# "https://github.com/vllm-project/vllm/blob/a377f0bd5e1fa0ca069e3dbf28f4de5af64d0bb1/vllm/config.py#L1160"
+# "https://github.com/vllm-project/vllm/blob/2ac6d0e75bc846998da56b50bf4f8853cb36d484/vllm/config.py#L1870"
```
Comment (gavrissh) [inline python/huggingfaceserver/huggingfaceserver/utils.py:27]: ~~I believe you can consume these methods through the vLLM pkg now~~

HFModel does not consume vLLM pkg.

**Comment by Datta0** [comment]
@yuzisun @sivanantha321  can you please take a look at this?

File: python/huggingfaceserver/huggingfaceserver/utils.py (hunk 1 above)
Comment (yuzisun) [reply python/huggingfaceserver/huggingfaceserver/utils.py:27]: I have the same question if we can reuse this function from vllm ?  It is hard to get keep up to date as vLLM frequently changes this

Comment (Datta0) [reply python/huggingfaceserver/huggingfaceserver/utils.py:27]: As of now, we don't mandate installation/existence of vLLM package. Its an extra and if one chooses to run on CPU or just via HuggingfaceGenerativeModel, it'd be an unnecessary.
Do we want to make vLLM mandataory?

**Pull request merged**